core/src/main/scala/io/circe/sbt/CirceIoPlugin.scala

```diff
@@ -0,0 +1,85 @@
+package io.circe.sbt
+
+import scoverage._
+import org.scalastyle.sbt._
+import org.typelevel.sbt._
+import org.typelevel.sbt.gha._
+import sbt.Keys._
+import sbt._
+
+import GenerativeKeys._
+import TypelevelKernelPlugin._
+import autoImport._
+import TypelevelCiPlugin.autoImport._
+import TypelevelSonatypePlugin.autoImport._
+import ScalastylePlugin.autoImport._
+import ScoverageSbtPlugin.autoImport._
+
+object CirceIoPlugin extends AutoPlugin {
+  object autoImport
+
+  override def trigger = allRequirements
+
+  override def requires = TypelevelCiPlugin && TypelevelSonatypePlugin && ScalastylePlugin && ScoverageSbtPlugin
+
+  override def buildSettings =
+    publishSettings ++ organizationSettings ++ githubActionsSettings
+
+  override def projectSettings = Seq(
+    // TODO: look over projects and see what's in common
+  )
+
+  lazy val publishSettings: Seq[Setting[_]] =
+    Seq(
+      tlSonatypeUseLegacyHost := true
+    )
```

Comment:
Would be awesome to migrate Circe to s01, then we can flip the switch here.

Reply:
That's the intention of setting it here!